bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -594,6 +594,17 @@ impl Room {
             .map_err(|e| ClientError::Generic { msg: e.to_string() })?;
         Ok(())
     }
+
+    pub async fn suggested_role_for_user(
+        &self,
+        user_id: String,
+    ) -> Result<RoomMemberRole, ClientError> {
+        let user_id = UserId::parse(&user_id)?;
+        self.inner
+            .get_suggested_user_role(user_id)
+            .await
+            .map_err(|e| ClientError::Generic { msg: e.to_string() })
```

Comment:
Does this work instead of doing the manual conversion to `ClientError::Generic`? It would be slightly more idiomatic.

```rust
Ok(self.inner
            .get_suggested_user_role(user_id)
            .await?)
```

Reply:
Done in https://github.com/matrix-org/matrix-rust-sdk/pull/3170/commits/307e5bf0333c5b45be4963efaa9ef5ff2c09902d. Thanks for the suggestion, I didn't realise it would be automatically mapped.